angular/projects/spark-angular/schematics/ng-add/index.ts

```diff
@@ -0,0 +1,179 @@
+ *  - Adds the Spark stylesheets to the project style file
+ *  - Adds the Spark class to the project html tag
+ */
+export function ngAdd(options: Schema): Rule {
+  return chain([
+    addSparkDependencies(options),
+    addSparkModule(options),
+    addAnimationsModule(options),
+    addSparkStyles(options),
+    addSparkClass(options),
+  ]);
+}
+
+/**
+ * Adds the spark and spark-angular dependencies to package.json
+ */
+function addSparkDependencies(_options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    // Schematics automatically adds to "dependencies" but it should be "devDependencies"
+    removePackageJsonDependency(tree, '@sparkdesignsystem/spark-angular');
+
+    const dependencies: NodeDependency[] = [
+      { name: '@sparkdesignsystem/spark', overwrite: true, type: NodeDependencyType.Dev, version: '^13.0.0' },
+      { name: '@sparkdesignsystem/spark-angular', overwrite: true, type: NodeDependencyType.Dev, version: '^10.0.0' },
+    ];
+
+    dependencies.forEach((dependency) => addPackageJsonDependency(tree, dependency));
+
+    context.addTask(new NodePackageInstallTask());
+
+    return tree;
+  };
+}
+
+/**
+ * Adds the SparkAngularModule to the root module of the specified project
+ */
+function addSparkModule(options: Schema): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const workspace = getWorkspace(tree);
+    const project = getProjectFromWorkspace(workspace, options.project);
+
+    addModuleImportToRootModule(tree, sparkAngularModuleName, sparkAngularModuleImport, project);
+
+    return tree;
+  };
+}
+
+/**
+ * Adds the BrowserAnimationsModule to the root module of the specified project
+ */
+function addAnimationsModule(options: Schema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const workspace = getWorkspace(tree);
+    const project = getProjectFromWorkspace(workspace, options.project);
+    const appModulePath = getAppModulePath(tree, getProjectMainFile(project));
+
+    // In case the project explicitly uses the NoopAnimationsModule, we should print a warning
+    // message that makes the user aware of the fact that we won't automatically set up
+    // animations. If we would add the BrowserAnimationsModule while the NoopAnimationsModule
+    // is already configured, we would cause unexpected behavior and runtime exceptions.
+    if (hasNgModuleImport(tree, appModulePath, noopAnimationsModuleName)) {
+      context.logger.error(
+        `Could not set up "${browserAnimationsModuleName}" ` +
+        `because "${noopAnimationsModuleName}" is already imported.`,
+      );
+      context.logger.info(`Please manually set up browser animations.`);
+      return;
+    }
+
+    addModuleImportToRootModule(tree, browserAnimationsModuleName, browserAnimationsModuleImport, project);
+
+    return tree;
+  };
+}
+
+/**
+ * Adds the Spark stylesheets to the project style file
+ */
+function addSparkStyles(options: Schema): Rule {
```

Comment:
Lets use arrow functions

Reply:
I would personally prefer not do this. By keeping these as functions, the implementation details can be hidden lower down in the file while the technical concerns can be more easily visible. You can currently glance at the file and very quickly see that ngAdd does this:

```
export function ngAdd(options: Schema): Rule {
  return chain([
    addSparkDependencies(options),
    addSparkModule(options),
    addAnimationsModule(options),
    addSparkStyles(options),
    addSparkClass(options),
  ]);
}
```

By making this change, these lines would have to go at the very end of the file so that we aren't using the variables before their declaration. I'll make this change if you want, but just explaining why I don't like it.

Reply:
Those familiar with JS would understand where to look to find the code they need to read and you also added a bunch of nice code commenting that helps. We should be using modern JS.